#!/bin/bash

#Exit script if any command fails
set -e

#Populate FASTQ array with prefixes of files to process
FASTQ=(190626_I114_FCH7TVNBBXY_L2_20-N4 190626_I114_FCH7TVNBBXY_L2_20-S1 190626_I114_FCH7TVNBBXY_L2_20-S3 190626_I114_FCH7TVNBBXY_L2_20-S4 190626_I114_FCH7TVNBBXY_L2_5-N1 190626_I114_FCH7TVNBBXY_L2_5-N2 190626_I114_FCH7TVNBBXY_L2_5-S3 190626_I114_FCH7TVNBBXY_L2_5-S4 190626_I114_FCH7TVNBBXY_L2_OC-S1 190626_I114_FCH7TVNBBXY_L3_20-N2 190626_I114_FCH7TVNBBXY_L3_5-N3 190626_I114_FCH7TVNBBXY_L3_5-S2 190626_I114_FCH7TVNBBXY_L3_OC-N1 190626_I114_FCH7TVNBBXY_L3_OC-N2 190626_I114_FCH7TVNBBXY_L3_OC-N4 190626_I114_FCH7TVNBBXY_L3_OC-N5 190626_I114_FCH7TVNBBXY_L3_OC-S2 190626_I114_FCH7TVNBBXY_L3_OC-S3 190626_I114_FCH7TVNBBXY_L4_20-N1 190626_I114_FCH7TVNBBXY_L4_20-S2 190626_I114_FCH7TVNBBXY_L4_5-S1 190626_I114_FCH7TVNBBXY_L4_OC-S5)

#Filter methylation data
#List of VCFs from BAT_calling
#Output directory and prefixes
#Methylation context
#Minimum and maximum reads
#Minimum methylation rate

for f in "${FASTQ[@]}"
do
  BAT_filter_vcf \
  --vcf ${CALLED}/${f}.vcf.gz \
  --out ${FILTERED}/${f}_CG.vcf.gz \
  --context CG \
  --MDP_min 10 --MDP_max 500
done
