---
title: "Methylation Landscape Analysis"
author: "Yaamini Venkataraman"
date: "2022-10-21"
output: html_document
---

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/yaaminivenkataraman/Documents/killifish-hypoxia-RRBS/output/05-analysis/methylation-landscape/") #Set root directory
```

#Install packages

```{r}
#install.packages("tidyverse")
#install.packages("RColorBrewer")
#install.packages("cowplot")
#install.packages("circlize")
require(tidyverse)
require(RColorBrewer)
require(cowplot)
require(circlize)
```

```{r}
sessionInfo()
```

# Create color scheme

```{r}
plotColors <- c(brewer.pal(5, "GnBu"), brewer.pal(5, "BuGn"))
plotColors <- plotColors[c(4,5,9,10)]
```

# Percent methylation by population and treatment

## Import and format data

```{r}
NavgMeth <- read.delim("N-samples-averages.bedgraph", header = TRUE, sep = "\t") %>%
  select(., -1) #Import data and remove first column
head(NavgMeth) #Confirm import
```

```{r}
NavgMethHyp <- read.delim("../summarize/20_5_N/N_mean_HY.bedgraph", header = TRUE, sep = "\t") %>%
  rename(., chr = X.chr) #Average methylation for hypoxia samples
head(NavgMethHyp)
```

```{r}
NavgMethNorm <- read.delim("../summarize/20_5_N/N_mean_NO.bedgraph", header = TRUE, sep = "\t") %>%
  rename(., chr = X.chr) #Average methylation for normoxia samples
head(NavgMethNorm)
```

```{r}
SavgMeth <- read.delim("S-samples-averages.bedgraph", header = TRUE, sep = "\t") %>%
  select(., -1) #Import data and remove first column
head(SavgMeth) #Confirm import
```

```{r}
SavgMethHyp <- read.delim("../summarize/20_5_S/S_mean_HY.bedgraph", header = TRUE, sep = "\t") %>%
  rename(., chr = X.chr) #Average methylation for hypoxia samples
head(SavgMethHyp)
```

```{r}
SavgMethNorm <- read.delim("../summarize/20_5_S/S_mean_NO.bedgraph", header = TRUE, sep = "\t") %>%
  rename(., chr = X.chr) #Average methylation for normoxia samples
head(SavgMethNorm)
```

```{r}
avgMethTreat <- full_join(x = NavgMethHyp, y = NavgMethNorm, by = c("chr", "start", "end")) %>%
  rename(mean_NBH_HY = mean_HY) %>%
  rename(mean_NBH_NO = mean_NO) %>%
  full_join(x = ., y = SavgMethHyp, by = c("chr", "start", "end")) %>%
  rename(mean_SC_HY = mean_HY) %>%
  full_join(x = ., y = SavgMethNorm, by = c("chr", "start", "end")) %>%
  rename(mean_SC_NO = mean_NO) #Combine all data using full_join to account for positions that are not common across populations
nrow(avgMethTreat) #Total number positions
head(avgMethTreat)
```

## Plot data

```{r}
ggplot(NavgMethHyp) + geom_density(mapping = aes(x = mean_HY, color = "NBH Hypoxia", fill = "NBH Hypoxia"), alpha = 0.1) +
  geom_density(mapping = aes(x = NavgMethNorm$mean_NO, color = "NBH Normoxia", fill = "NBH Normoxia"), alpha = 0.1) +
  scale_y_continuous(limits = c(0,10)) +
  labs(x = "Proportion CpG Methylation", y = "Density") +
  scale_color_manual(name = "Population and Treatment",
                     breaks = c("NBH Hypoxia", "NBH Normoxia"),
                     values = c("NBH Hypoxia" = plotColors[2],
                                "NBH Normoxia" = plotColors[1]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))
```

```{r}
ggplot(SavgMethHyp) + geom_density(mapping = aes(x = mean_HY, color = "SC Hypoxia", fill = "SC Hypoxia"), alpha = 0.1) +
  geom_density(mapping = aes(x = SavgMethNorm$mean_NO, color = "SC Normoxia", fill = "SC Normoxia"), alpha = 0.1) +
  scale_y_continuous(limits = c(0,25)) +
  labs(x = "Proportion CpG Methylation", y = "Density") +
  scale_color_manual(name = "Population and Treatment",
                     breaks = c("SC Hypoxia", "SC Normoxia"),
                     values = c("SC Hypoxia" = plotColors[4],
                                "SC Normoxia" = plotColors[3]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))
```
```{r}
#Create main plot
#Create inset plot
#Plot together

mainPlot <- ggplot(avgMethTreat) + geom_density(mapping = aes(x = mean_NBH_HY, color = "NBH Hypoxia", fill = "NBH Hypoxia"), alpha = 0.1) +
  geom_density(mapping = aes(x = mean_NBH_NO, color = "NBH Normoxia", fill = "NBH Normoxia"), alpha = 0.1) +
  geom_density(mapping = aes(x = mean_SC_HY, color = "SC Hypoxia", fill = "SC Hypoxia"), alpha = 0.1) +
  geom_density(mapping = aes(x = mean_SC_HY, color = "SC Normoxia", fill = "SC Normoxia"), alpha = 0.1) +
  scale_y_continuous(limits = c(0,25)) +
  labs(x = "Proportion CpG Methylation", y = "Density") +
  scale_color_manual(name = "Population and Treatment",
                     breaks = c("NBH Hypoxia", 
                                "NBH Normoxia", 
                                "SC Hypoxia", 
                                "SC Normoxia"),
                     values = c("NBH Hypoxia" = plotColors[2],
                                "NBH Normoxia" = plotColors[1],
                                "SC Hypoxia" = plotColors[4],
                                "SC Normoxia" = plotColors[3]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))

insetPlot <- mainPlot +
  scale_x_continuous(limits = c(0,0.25)) +
  scale_y_continuous(limits = c(0,10)) +
  theme_classic() + theme(legend.position = "none")

methWithInset <- ggdraw() +
  draw_plot(mainPlot) +
  draw_plot(insetPlot, x = 0.25, y = 0.5, width = 0.5, height = 0.5)

methWithInset

ggsave("average-methylation-density-plot.pdf", methWithInset, height = 8.5, width = 11, dpi = 300)
```

# Methylation difference by population and treatment

## Import and format data

```{r}
NdiffMeth <- read.delim("../summarize/20_5_N/N_diff_NO_HY.bedgraph", header = TRUE) %>%
  rename(chr = X.chr) %>%
  mutate(diff_HY_minus_NO_NBH = -1 * diff_NO_minus_HY) %>%
  select(-diff_NO_minus_HY) #Import data, rename column, create a new column with HY-NO meth diff, and drop old meth diff column
head(NdiffMeth)
```

```{r}
SdiffMeth <- read.delim("../summarize/20_5_S/S_diff_NO_HY.bedgraph", header = TRUE) %>%
  rename(chr = X.chr) %>%
  mutate(diff_HY_minus_NO_SC = -1 * diff_NO_minus_HY) %>%
  select(-diff_NO_minus_HY) #Import data, rename column, create a new column with HY-NO meth diff, and drop old meth diff column
head(SdiffMeth)
```

```{r}
diffMethTreat <- full_join(x = NdiffMeth, y = SdiffMeth, by = c("chr", "start", "end")) #Combine all data using full_join to account for positions that are not common across populations
nrow(diffMethTreat) #Total number positions
head(diffMethTreat)
```

## Plot data

```{r}
ggplot(NdiffMeth) + geom_density(mapping = aes(x = diff_HY_minus_NO_NBH, color = "NBH", fill = "NBH"), alpha = 0.1) +
  labs(x = "Methylation Difference (HY - NO)", y = "Density") +
  scale_y_continuous(limits = c(0,45)) +
  scale_color_manual(name = "Population",
                     breaks = c("NBH"),
                     values = c("NBH" = plotColors[2]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))
```

```{r}
ggplot(SdiffMeth) + geom_density(mapping = aes(x = diff_HY_minus_NO_SC, color = "SC", fill = "SC"), alpha = 0.1) +
  labs(x = "Methylation Difference (HY - NO)", y = "Density") +
  scale_y_continuous(limits = c(0,45)) +
  scale_color_manual(name = "Population",
                     breaks = c("SC"),
                     values = c("SC" = plotColors[4]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))
```

```{r}
#Create main plot
#Create inset plot
#Plot together

mainPlotMethDiff <- ggplot(diffMethTreat) + geom_density(mapping = aes(x = diff_HY_minus_NO_NBH, color = "NBH", fill = "NBH"), alpha = 0.1) +
  geom_density(mapping = aes(x = diff_HY_minus_NO_SC, color = "SC", fill = "SC"), alpha = 0.1) +
  labs(x = "Methylation Difference (HY - NO)", y = "Density") +
  scale_color_manual(name = "Population",
                     breaks = c("NBH", "SC"),
                     values = c("NBH" = plotColors[2],
                                "SC" = plotColors[4]),
                     aesthetics = c("color", "fill")) +
  theme_classic() + theme(axis.text.x = element_text(size = 13),
                          axis.title.x = element_text(size = 20),
                          axis.text.y = element_text(size = 13),
                          axis.title.y = element_text(size = 20))

insetPlotMethDiff <- mainPlotMethDiff +
  scale_x_continuous(limits = c(-0.25,0.25)) +
  scale_y_continuous(limits = c(0,10)) +
  theme(legend.position = "none")

methDiffWithInset <- ggdraw() +
  draw_plot(mainPlotMethDiff) +
  draw_plot(insetPlotMethDiff, x = 0.5, y = 0.65, width = 0.5, height = 0.3)

methDiffWithInset

ggsave("diff-methylation-density-plot.pdf", methDiffWithInset, height = 8.5, width = 11, dpi = 300)
```

# Chromosomal positions of methylation by population

## Import and format data

```{r}
methByChrNBH <- avgMethTreat %>%
  filter(mean_NBH_HY != "NA") %>%
  group_by(., chr) %>%
  summarize(., n = n()) %>%
  arrange(., desc(n)) #Remove rows without NBH data, group by chromosome, count the number of methylated positions with data per chromosome, and arrange in order of chromosomes with most to least data
head(methByChrNBH)
```

```{r}
methByChrSC <- avgMethTreat %>%
  filter(mean_SC_HY != "NA") %>%
  group_by(., chr) %>%
  summarize(., n = n()) %>%
  arrange(., desc(n))
head(methByChrSC)
```

I'm going to plot data for four chromosomes: KN805526.1 and KN805525.1 (chr with most data for NBH) and KN811310.1 and KN811340.1 (chr with most data for SC).

```{r}
NavgMethHypCircos <- NavgMethHyp %>%
  filter(chr == c("KN805525.1", "KN805526.1", "KN811310.1", "KN811340.1"))
NavgMethNormCircos <- NavgMethNorm %>%
  filter(chr == c("KN805525.1", "KN805526.1", "KN811310.1", "KN811340.1"))
head(NavgMethHypCircos)
head(NavgMethNormCircos)
```

```{r}
SavgMethHypCircos <- SavgMethHyp %>%
  filter(chr == c("KN805525.1", "KN805526.1", "KN811310.1", "KN811340.1"))
SavgMethNormCircos <- SavgMethNorm %>%
  filter(chr == c("KN805525.1", "KN805526.1", "KN811310.1", "KN811340.1"))
head(SavgMethHypCircos)
head(SavgMethNormCircos)
```

```{r}
chrLength <- read.delim("../../../genome-features/mummichog.chrom.length", header = FALSE, col.names = c("chr", "length")) #Import chr length information
head(chrLength)
```

## Plot data

Relevant tutorials:

- https://jokergoo.github.io/circlize_book/book/
- https://www.royfrancis.com/beautiful-circos-plots-in-r/

```{r}
#pdf("N-S-circos.pdf", width = 11, height = 8.5)


circos.clear() #Create a blank circos plot
col_text <- "grey20" #Add text color
circos.par(gap.degree = 10,
           cell.padding = c(0, 0, 0, 0)) #par information for plot
circos.initialize(factors = c("KN805525.1", "KN805526.1", "KN811310.1", "KN811340.1"), 
                  xlim = matrix(c(rep(0, 4), 
                                6356336, 6250690, 5246820, 5018176), 
                              ncol = 2)) #Initialize circos

# Add chromosome delinations
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  chr = CELL_META$sector.index
  xlim = CELL_META$xlim
  ylim = CELL_META$ylim
  circos.text(mean(xlim), mean(ylim), chr, cex = 0.75, col = col_text, 
              facing = "bending.inside", niceFacing = TRUE)
}, bg.col = "grey90", bg.border = FALSE, track.height = 0.06)

# Genome x axis
brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7)*10^6 #Set axis breaks
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
  circos.axis(h = "top", major.at = brk, labels = round(brk/10^6, 1), labels.cex = 0.75, 
              col = col_text, labels.col = col_text, lwd = 0.7, labels.facing = "clockwise")
}, bg.border = FALSE)

# NBH Normoxia
circos.track(factors = NavgMethNormCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  value = NavgMethNormCircos$mean_NO
  pos = NavgMethNormCircos$start
  circos.barplot(value, pos,
                 border = plotColors[1], col = plotColors[1])
}, track.height = 0.15, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

circos.yaxis(at = c(0, 0.5, 1), 
             labels.cex = 0.75, lwd = 0, tick.length = 0, labels.col = col_text, col = "#FFFFFF") #Add yaxis

# NBH Hypoxia
circos.track(factors = NavgMethHypCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  value = NavgMethHypCircos$mean_HY
  pos = NavgMethHypCircos$start
circos.barplot(value, pos, border = plotColors[2], col = plotColors[2])
}, track.height = 0.15, bg.border = FALSE)

circos.yaxis(at = c(0, 0.5, 1), 
             labels.cex = 0.75, lwd = 0, tick.length = 0, labels.col = col_text, col = "#FFFFFF") #Add yaxis

# SC Normoxia
circos.track(factors = SavgMethNormCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  value = SavgMethNormCircos$mean_NO
  pos = SavgMethNormCircos$start
  circos.barplot(value, pos,
                 border = plotColors[3], col = plotColors[3])
}, track.height = 0.15, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

circos.yaxis(at = c(0, 0.5, 1), 
             labels.cex = 0.75, lwd = 0, tick.length = 0, labels.col = col_text, col = "#FFFFFF") #Add yaxis

# SC Hypoxia
circos.track(factors = SavgMethHypCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  value = SavgMethHypCircos$mean_HY
  pos = SavgMethHypCircos$start
circos.barplot(value, pos, border = plotColors[4], col = plotColors[4])
}, track.height = 0.15, bg.border = FALSE)

circos.yaxis(at = c(0, 0.5, 1), 
             labels.cex = 0.75, lwd = 0, tick.length = 0, labels.col = col_text, col = "#FFFFFF") #Add yaxis

#dev.off()
```

```{r}
write.table(NavgMethNormCircos, file = "NavgMethNormCircos.tab", quote = FALSE, sep = "\t", col.names = TRUE, row.names = FALSE)
```


# TO DO

- Fix points out of plotting sector