---
title: "Methylation Landscape Analysis"
author: "Yaamini Venkataraman"
date: "2022-10-21"
output: html_document
---

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "/Users/yaaminivenkataraman/Documents/killifish-hypoxia-RRBS/output/05-analysis/new-genome/methylation-landscape/") #Set root directory
```

#Install packages

```{r}
#install.packages("tidyverse")
#install.packages("RColorBrewer")
#install.packages("cowplot")
#install.packages("circlize")
require(tidyverse)
require(RColorBrewer)
require(cowplot)
require(circlize)
```

```{r}
sessionInfo()
```

# Create color scheme

```{r}
plotColors <- c(brewer.pal(9, "GnBu"), 
                brewer.pal(9, "BuGn")) #Create color scheme with greys (outside controls), green-blues, and blue-greens
plotColors <- plotColors[c(9,8,6,18,16,14)] #Pick colors to use in figures
```

# Import data

```{r}
avgMethUnion <- read.delim("missing_1/all-samples-averages-union.bedgraph", header = TRUE, sep = "\t", row.names = 1, na.strings = "N/A") #Import union bedgraph with average methylation across all positions
head(avgMethUnion) #Confirm import
```

```{r}
avgMethCommonLoci <- read.delim("missing_1/all-common-loci-averages.bedgraph", header = TRUE, sep = "\t", row.names = 1, na.strings = "N/A") #Import bedgraph with common loci across all samples
head(avgMethCommonLoci) #Confirm import
```

```{r}
chrLength <- read.delim("../../../../genome-features/mummichog.chrom.length", header = FALSE, col.names = c("chr", "length")) #Import chr length information
head(chrLength)
```

# Chromosomal positions of methylation by population

I will create a circos plot to visualize methylation at common positions across populations and treatments.

## Format data

To format the data for the plot, I need to know which chromosomes to visualize. I'll choose chromosomes based on the positions of important genes such as HIF-1a and AHR.

```{bash}
grep "hypoxia" ../../../../genome-features/GCF_011125445.2_MU-UCD_Fhet_4.1_cds_from_genomic.fna
```

```{bash}
grep "aryl hydrocarbon receptor" ../../../../genome-features/GCF_011125445.2_MU-UCD_Fhet_4.1_cds_from_genomic.fna
```

HIF-1a is in NC_046379.1 and AHR is in NC_046384.1, so I'll plot these chromosomes.

```{r}
#Select relevant columns and filter out the two chromosomes with HIF and AHR.

NavgMethOCCircos <- avgMethCommonLoci %>%
  select(., chr, pos, NBH.OC.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
NavgMethNOCircos <- avgMethCommonLoci %>%
  select(., chr, pos, NBH.NO.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
NavgMethHYCircos <- avgMethCommonLoci %>%
  select(., chr, pos, NBH.HY.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
head(NavgMethOCCircos)
head(NavgMethNOCircos)
head(NavgMethHYCircos)
```

```{r}
#Select relevant columns and filter out the two chromosomes with HIF and AHR.

SavgMethOCCircos <- avgMethCommonLoci %>%
  select(., chr, pos, SC.OC.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
SavgMethNOCircos <- avgMethCommonLoci %>%
  select(., chr, pos, SC.NO.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
SavgMethHYCircos <- avgMethCommonLoci %>%
  select(., chr, pos, SC.HY.average) %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
head(SavgMethOCCircos)
head(SavgMethNOCircos)
head(SavgMethHYCircos)
```

```{r}
chrLength %>%
  filter(chr == c("NC_046379.1", "NC_046384.1"))
```

## Plot data

Relevant tutorials:

- https://jokergoo.github.io/circlize_book/book/
- https://www.royfrancis.com/beautiful-circos-plots-in-r/

```{r}
pdf("N-S-circos.pdf", width = 11, height = 8.5)

circos.clear() #Create a blank circos plot
col_text <- "grey20" #Add text color
circos.par(gap.degree = 10,
           cell.padding = c(0, 0, 0, 0)) #par information for plot
circos.initialize(factors = c("NC_046379.1", "NC_046384.1"), 
                  xlim = matrix(c(rep(0, 2), 
                                  33976892, 28860465), 
                                ncol = 2)) #Initialize circos

# Add chromosome delinations
circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
  chr = CELL_META$sector.index
  xlim = CELL_META$xlim
  ylim = CELL_META$ylim
  circos.text(mean(xlim), mean(ylim), chr, cex = 0.75, col = col_text, 
              facing = "bending.inside", niceFacing = TRUE)
}, bg.col = "grey90", bg.border = FALSE, track.height = 0.06)

# Genome x axis
brk <- c(0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5)*10^7 #Set axis breaks
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
  circos.axis(h = "top", major.at = brk, labels = round(brk/10^7, 1), labels.cex = 1, 
              col = col_text, labels.col = col_text, lwd = 1, labels.facing = "clockwise")
}, bg.border = FALSE)

# NBH Outside Control
circos.track(factors = NavgMethOCCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = NavgMethOCCircos$chr == CELL_META$sector.index
  value = NavgMethOCCircos$NBH.OC.average[l]
  pos = NavgMethOCCircos$pos[l]
  circos.barplot(value, pos,
                 border = plotColors[1], col = plotColors[1])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

circos.yaxis(at = c(0, 0.5, 1), 
             labels.cex = 1, lwd = 1, tick.length = 1, labels.col = col_text, col = "grey20") #Add y-axis

# NBH Normoxia
circos.track(factors = NavgMethNOCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = NavgMethNOCircos$chr == CELL_META$sector.index
  value = NavgMethNOCircos$NBH.NO.average[l]
  pos = NavgMethNOCircos$pos[l]
  circos.barplot(value, pos,
                 border = plotColors[2], col = plotColors[2])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

# NBH Hypoxia
circos.track(factors = NavgMethHYCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = NavgMethHYCircos$chr == CELL_META$sector.index
  value = NavgMethHYCircos$NBH.HY.average[l]
  pos = NavgMethHYCircos$pos[l]
  circos.barplot(value, pos, border = plotColors[3], col = plotColors[3])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

# SC Outside Control
circos.track(factors = SavgMethOCCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = SavgMethOCCircos$chr == CELL_META$sector.index
  value = SavgMethOCCircos$SC.OC.average[l]
  pos = SavgMethOCCircos$pos[l]
  circos.barplot(value, pos,
                 border = plotColors[4], col = plotColors[4])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

# SC Normoxia
circos.track(factors = SavgMethNOCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = SavgMethNOCircos$chr == CELL_META$sector.index
  value = SavgMethNOCircos$SC.NO.average[l]
  pos = SavgMethNOCircos$pos[l]
  circos.barplot(value, pos,
                 border = plotColors[5], col = plotColors[5])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

# SC Hypoxia
circos.track(factors = SavgMethHYCircos$chr, ylim = c(0,1), panel.fun=function(x, y) {
  l = SavgMethHYCircos$chr == CELL_META$sector.index
  value = SavgMethHYCircos$SC.HY.average[l]
  pos = SavgMethHYCircos$pos[l]
  circos.barplot(value, pos, border = plotColors[6], col = plotColors[6])
}, track.height = 0.13, bg.border = FALSE) #Add a barplot specifying chr as a factor (sector), and ylim. Add mean as values for barplot and start bp as pos.

dev.off()
```

# Number of methylated and unmethylated CpGs

## Format data



# Percent methylation by population and treatment

## Format data

```{r}
avgMethTreatLong <- avgMethUnion %>% 
  select(., chrom, start, 
         NBH.OC.average, NBH.NO.average, NBH.HY.average,
         SC.OC.average, SC.NO.average, SC.HY.average) %>%
  pivot_longer(., cols = 3:8, names_to = "treatment") %>%
  mutate(., population = gsub(x = treatment, pattern = ".OC.average", replacement = "")) %>%
  mutate(., population = gsub(x = population, pattern = ".NO.average", replacement = "")) %>%
  mutate(., population = gsub(x = population, pattern = ".HY.average", replacement = "")) #Take union data and select relevant columns. Pivot longer so treatment x population information is in a separate column. Create a new column with population using a series of gsub commands.
head(avgMethTreatLong) #Confirm formatting
```

## Plot data

```{r}
ggplot(avgMethTreatLong) +
  geom_density(mapping = aes(x = value, color = treatment, linetype = treatment)) +
  labs(x = "Proportion CpG Methylation", y = "Density") +
  scale_color_manual(name = "Population and Treatment",
                     breaks = c("NBH.OC.average", 
                                "NBH.NO.average", 
                                "NBH.HY.average",
                                "SC.OC.average", 
                                "SC.NO.average", 
                                "SC.HY.average"),
                     labels = c("NBH Outside Control",
                                "NBH Normoxia",
                                "NBH Hypoxia",
                                "SC Outside Control",
                                "SC Normoxia",
                                "SC Hypoxia"),
                     values = c("NBH.OC.average" = plotColors[1], 
                                "NBH.NO.average" = plotColors[2], 
                                "NBH.HY.average" = plotColors[3],
                                "SC.OC.average" = plotColors[4], 
                                "SC.NO.average" = plotColors[5], 
                                "SC.HY.average" = plotColors[6])) +
  scale_linetype_manual(name = "Population and Treatment",
                        breaks = c("NBH.OC.average", 
                                "NBH.NO.average", 
                                "NBH.HY.average",
                                "SC.OC.average", 
                                "SC.NO.average", 
                                "SC.HY.average"),
                     labels = c("NBH Outside Control",
                                "NBH Normoxia",
                                "NBH Hypoxia",
                                "SC Outside Control",
                                "SC Normoxia",
                                "SC Hypoxia"),
                     values = c("NBH.OC.average" = 1, 
                                "NBH.NO.average" = 1, 
                                "NBH.HY.average" = 1,
                                "SC.OC.average" = 2, 
                                "SC.NO.average" = 2, 
                                "SC.HY.average" = 2),
                     aesthetics = c("linetype")) +
  theme_classic(base_size = 13) + theme(axis.text.x = element_text(size = 13),
                                        axis.title.x = element_text(size = 20),
                                        axis.text.y = element_text(size = 13),
                                        axis.title.y = element_text(size = 20))
```

```{r}
#Create main plot
#Create inset plot
#Plot together

mainPlot <- ggplot(avgMethTreatLong) +
  geom_density(mapping = aes(x = value, color = treatment, linetype = treatment)) +
  labs(x = "Proportion CpG Methylation", y = "Density") +
  scale_color_manual(name = "Population and Treatment",
                     breaks = c("NBH.OC.average", 
                                "NBH.NO.average", 
                                "NBH.HY.average",
                                "SC.OC.average", 
                                "SC.NO.average", 
                                "SC.HY.average"),
                     labels = c("NBH Outside Control",
                                "NBH Normoxia",
                                "NBH Hypoxia",
                                "SC Outside Control",
                                "SC Normoxia",
                                "SC Hypoxia"),
                     values = c("NBH.OC.average" = plotColors[1], 
                                "NBH.NO.average" = plotColors[2], 
                                "NBH.HY.average" = plotColors[3],
                                "SC.OC.average" = plotColors[4], 
                                "SC.NO.average" = plotColors[5], 
                                "SC.HY.average" = plotColors[6])) +
  scale_linetype_manual(name = "Population and Treatment",
                        breaks = c("NBH.OC.average", 
                                "NBH.NO.average", 
                                "NBH.HY.average",
                                "SC.OC.average", 
                                "SC.NO.average", 
                                "SC.HY.average"),
                     labels = c("NBH Outside Control",
                                "NBH Normoxia",
                                "NBH Hypoxia",
                                "SC Outside Control",
                                "SC Normoxia",
                                "SC Hypoxia"),
                     values = c("NBH.OC.average" = 1, 
                                "NBH.NO.average" = 1, 
                                "NBH.HY.average" = 1,
                                "SC.OC.average" = 2, 
                                "SC.NO.average" = 2, 
                                "SC.HY.average" = 2),
                     aesthetics = c("linetype")) +
  theme_classic(base_size = 13) + theme(axis.text.x = element_text(size = 13),
                                        axis.title.x = element_text(size = 20),
                                        axis.text.y = element_text(size = 13),
                                        axis.title.y = element_text(size = 20))

insetPlot <- mainPlot +
  scale_x_continuous(limits = c(0,0.25)) +
  scale_y_continuous(limits = c(0,3)) +
  theme_classic() + theme(legend.position = "none")

methWithInset <- ggdraw() +
  draw_plot(mainPlot) +
  draw_plot(insetPlot, x = 0.2, y = 0.5, width = 0.5, height = 0.5)

methWithInset

ggsave("average-methylation-density-plot.pdf", methWithInset, height = 8.5, width = 11, dpi = 300)
```

